platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -68,20 +84,31 @@ public GossipEvent(final BaseEventHashedData hashedData, final BaseEventUnhashed
      */
     @Override
     public void serialize(final SerializableDataOutputStream out) throws IOException {
-        out.writeSerializable(hashedData, false);
-        out.writeSerializable(unhashedData, false);
+        if (serializedVersion < ClassVersion.ROSTER_ROUND) {
+            out.writeSerializable(hashedData, false);
+            out.writeSerializable(unhashedData, false);
+        } else {
+            out.writeSerializable(hashedData, false);
+            out.writeByteArray(unhashedData.getSignature());
+        }
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void deserialize(final SerializableDataInputStream in, final int version) throws IOException {
-        hashedData = in.readSerializable(false, BaseEventHashedData::new);
-        unhashedData = in.readSerializable(false, BaseEventUnhashedData::new);
-        if (version == ClassVersion.ORIGINAL) {
-            in.readLong(); // roundCreated
+        serializedVersion = version;
+        if (version < ClassVersion.ROSTER_ROUND) {
+            hashedData = in.readSerializable(false, BaseEventHashedData::new);
+            unhashedData = in.readSerializable(false, BaseEventUnhashedData::new);
+        } else {
+            hashedData = in.readSerializable(false, BaseEventHashedData::new);
+            final byte[] signature = in.readByteArray(MAX_SIG_LENGTH);
+            unhashedData = new BaseEventUnhashedData(null, signature);
         }
+        // remove update of other parent event descriptor after 0.45.0 hits mainnet.
```

Comment:
Once 45 hits mainnet, would we be able to delete `BaseEventUnhashedData` entirely, and just directly use the signature?

Reply:
there is more refactor to perform to get rid of `BaseEventUnhashedData` so that nothing is referencing it.   But after this pr hits mainnet, there should no longer be instances of it in the serialization scheme and we're free to proceed with excising the tumor from the code.